File: frontend/src/types/pathBrowse.ts
export type PathBrowsePath = {
  name: string
  path: string
}

export type PathBrowseItem = {
  name: string
  path: string
  is_dir: boolean
  updated_at?: any
  size?: number | null
}

export type PathBrowseResponse = {
  dir_path: string
  exists: boolean
  paths: PathBrowsePath[]
  items: PathBrowseItem[]
  scanned_at: string
}

File: backend/app/schemas/path_browse.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class PathBrowseIn(BaseModel):
    path: str = Field(default="")
    refresh: bool = False
    max_items: int = Field(default=200, ge=1, le=5000)


class PathBrowsePathOut(BaseModel):
    name: str
    path: str


class PathBrowseItemOut(BaseModel):
    name: str
    path: str
    is_dir: bool
    updated_at: Any | None = None
    size: int | None = None


class PathBrowseOut(BaseModel):
    dir_path: str
    exists: bool
    paths: list[PathBrowsePathOut] = []
    items: list[PathBrowseItemOut] = []
    scanned_at: datetime = Field(default_factory=datetime.now)

